--- a/CMakeLists.txt	2026-03-21 18:28:51.026492996 -0400
+++ b/CMakeLists.txt	2026-03-21 18:33:49.324621541 -0400
@@ -9,11 +9,18 @@
 include(FetchContent)
 
 # FTXUI
-FetchContent_Declare(
-  ftxui
-  GIT_REPOSITORY https://github.com/ArthurSonzogni/FTXUI
-  GIT_TAG        v5.0.0
-)
+if (DEFINED ENV{FTXUI_SRC})
+  FetchContent_Declare(
+    ftxui
+    SOURCE_DIR "$ENV{FTXUI_SRC}"
+  )
+else()
+  FetchContent_Declare(
+    ftxui
+    GIT_REPOSITORY https://github.com/ArthurSonzogni/FTXUI
+    GIT_TAG        v5.0.0
+  )
+endif()
 set(FTXUI_BUILD_DOCS     OFF CACHE BOOL "" FORCE)
 set(FTXUI_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
 set(FTXUI_BUILD_TESTS    OFF CACHE BOOL "" FORCE)
@@ -45,11 +52,18 @@
 endif()
 
 # Catch2
-FetchContent_Declare(
-  Catch2
-  GIT_REPOSITORY https://github.com/catchorg/Catch2
-  GIT_TAG        v3.7.1
-)
+if (DEFINED ENV{CATCH2_SRC})
+  FetchContent_Declare(
+    Catch2
+    SOURCE_DIR "$ENV{CATCH2_SRC}"
+  )
+else()
+  FetchContent_Declare(
+    Catch2
+    GIT_REPOSITORY https://github.com/catchorg/Catch2
+    GIT_TAG        v3.7.1
+  )
+endif()
 FetchContent_MakeAvailable(Catch2)
 
 enable_testing()
